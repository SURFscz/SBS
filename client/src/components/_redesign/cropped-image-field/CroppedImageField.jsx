import React from "react";
import PropTypes from "prop-types";
import I18n from "../../../locale/I18n";
import {ReactComponent as NotFoundIcon} from "../../../icons/image-not-found.svg";
import "./CroppedImageField.scss";
import {isEmpty} from "../../../utils/Utils";
import "react-image-crop/dist/ReactCrop.css";
import Logo from "../logo/Logo";
import Button from "../../button/Button";
import CroppedImageDialog from "../cropped-image-dialog/CroppedImageDialog";
import ConfirmationDialog from "../../confirmation-dialog/ConfirmationDialog";
import ErrorIndicator from "../error-indicator/ErrorIndicator";

export default class CroppedImageField extends React.PureComponent {

    constructor(props, context) {
        super(props, context);
        this.state = {
            dialogOpen: false,
            confirmationDialogOpen: false,
            confirmationDialogAction: () => true
        }
    }

    onInternalChange = (val) => {
        this.setState({dialogOpen: false});
        const {onChange} = this.props;
        setTimeout(() => onChange(val), 425);
    }

    closeDialog = () => {
        this.setState({dialogOpen: false});
    }

    render() {
        const {dialogOpen, confirmationDialogOpen, confirmationDialogAction} = this.state;
        const {title, name, value, isNew, secondRow = false, initial = false, disabled = false, includeLogoGallery = false} = this.props;
        const imageRequired = !initial && isEmpty(value);
        return (
            <div className={`cropped-image-field ${secondRow ? "second-row" : ""}`}>
                <ConfirmationDialog isOpen={confirmationDialogOpen}
                                    cancel={() => this.setState({confirmationDialogOpen: false})}
                                    confirm={confirmationDialogAction}
                                    isWarning={true}
                                    question={I18n.t("forms.imageDeleteConfirmation")}/>
                <CroppedImageDialog onSave={this.onInternalChange}
                                    onCancel={this.closeDialog}
                                    isOpen={dialogOpen}
                                    isNew={isNew}
                                    name={name}
                                    value={value}
                                    includeLogoGallery={includeLogoGallery}
                                    title={title}/>
                <label className="info" htmlFor="">{title}<sup className="required">*</sup></label>
                <section className="file-upload">
                    {!value && <div className={`no-image ${imageRequired ? "error" : ""}`}>
                        {<NotFoundIcon/>}
                    </div>}
                    {value &&
                    <div className="preview">
                        {value && <Logo className="cropped-img" src={value}/>}
                    </div>}
                    <div className="file-upload-actions">
                        <Button txt={value ? I18n.t("forms.change") : I18n.t("forms.add")}
                                disabled={disabled}
                                small={true}
                                onClick={() => this.setState({dialogOpen: true})}/>
                    </div>
                </section>
                {imageRequired && <ErrorIndicator msg={I18n.t("forms.imageRequired")} />}
            </div>
        );
    }

}

CroppedImageField.propTypes = {
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    isNew: PropTypes.bool.isRequired,
    title: PropTypes.string,
    value: PropTypes.string,
    secondRow: PropTypes.bool,
    disabled: PropTypes.bool,
    initial: PropTypes.bool,
};
