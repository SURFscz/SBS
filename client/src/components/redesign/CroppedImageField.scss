@import "../../stylesheets/vars.scss";

.cropped-image-field {
    display: flex;
    flex-direction: column;
    margin-top: 15px;
    grid-column-start: second;
    grid-row-start: 1;
    grid-row-end: 99;

    &.second-row {
        grid-row-start: 2;
    }

    label.info {
        display: inline-block;
        margin: 12px 0 6px 0;
        min-width: 210px;
        color: black;
        position: relative;
        font-weight: bold;
    }

    .file-upload {
        display: flex;
        margin: 5px 0;
        flex-direction: column;
    }

    .preview {
        img.cropped-img {
            width: 240px;
            height: auto;
            border-radius: 4px;
        }
    }

    .no-image svg {
        width: 240px;
        height: auto;
        padding: 50px;
        border: 3px dotted $lighter-grey;
        fill: $lighter-grey;
    }

    .no-image.error svg {
        border-color: $error-red;
        fill: $error-red-background;
    }

    .file-upload-actions {
        display: flex;
        margin-top: 15px;

        a {
            max-width: 240px;
        }

    }

}
