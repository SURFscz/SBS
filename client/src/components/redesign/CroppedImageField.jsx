import React from "react";
import PropTypes from "prop-types";
import I18n from "i18n-js";
import {ReactComponent as NotFoundIcon} from "../../icons/image-not-found.svg";
import "./CroppedImageField.scss";
import {isEmpty} from "../../utils/Utils";
import "react-image-crop/lib/ReactCrop.scss";
import Logo from "./Logo";
import Button from "../Button";
import CroppedImageDialog from "./CroppedImageDialog";
import ConfirmationDialog from "../ConfirmationDialog";
import ErrorIndicator from "./ErrorIndicator";

export default class CroppedImageField extends React.PureComponent {

    constructor(props, context) {
        super(props, context);
        this.state = {
            dialogOpen: false,
            confirmationDialogOpen: false,
            confirmationDialogAction: () => true
        }
    }

    onInternalChange = (val) => {
        this.setState({dialogOpen: false});
        const {onChange} = this.props;
        setTimeout(() => onChange(val), 425);
    }

    closeDialog = () => this.setState({dialogOpen: false});

    render() {
        const {dialogOpen, confirmationDialogOpen, confirmationDialogAction} = this.state;
        const {title, name, value, secondRow = false, initial = false, disabled = false} = this.props;
        const imageRequired = !initial && isEmpty(value);
        return (
            <div className={`cropped-image-field ${secondRow ? "second-row" : ""}`}>
                <ConfirmationDialog isOpen={confirmationDialogOpen}
                                    cancel={() => this.setState({confirmationDialogOpen: false})}
                                    confirm={confirmationDialogAction}
                                    isWarning={true}
                                    question={I18n.t("forms.imageDeleteConfirmation")}/>
                <CroppedImageDialog onSave={this.onInternalChange} onCancel={this.closeDialog} isOpen={dialogOpen}
                                    name={name} value={value} title={title}/>
                <label className="info" htmlFor="">{title}</label>
                <section className="file-upload">
                    {!value && <div className={`no-image ${imageRequired ? "error" : ""}`}>
                        {<NotFoundIcon/>}
                    </div>}
                    {value &&
                    <div className="preview">
                        {value && <Logo className="cropped-img" src={value}/>}
                    </div>}
                    <div className="file-upload-actions">
                        <Button txt={value ? I18n.t("forms.change") : I18n.t("forms.add")}
                                disabled={disabled}
                                onClick={() => this.setState({dialogOpen: true})}/>
                    </div>
                </section>
                {imageRequired && <ErrorIndicator msg={I18n.t("forms.imageRequired")} />}
            </div>
        );
    }

}

CroppedImageField.propTypes = {
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    isNew: PropTypes.bool.isRequired,
    title: PropTypes.string,
    value: PropTypes.string,
    secondRow: PropTypes.bool,
    disabled: PropTypes.bool,
    initial: PropTypes.bool,
};
