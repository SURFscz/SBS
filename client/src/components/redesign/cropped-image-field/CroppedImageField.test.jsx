import React from "react";
import { render } from "@testing-library/react";
import CroppedImageField from "./CroppedImageField";

describe("CroppedImageField", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <CroppedImageField
                name="testImage"
                label="Test Image"
                value="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUA..."
                onChange={() => {}}
                isNew={false}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
