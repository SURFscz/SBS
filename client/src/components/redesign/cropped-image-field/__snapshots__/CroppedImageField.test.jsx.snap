// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CroppedImageField matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="cropped-image-field "
  >
    <label
      class="info"
      for=""
    >
      <sup
        class="required"
      >
        *
      </sup>
    </label>
    <section
      class="file-upload"
    >
      <div
        class="preview"
      >
        <img
          alt=""
          class="cropped-img"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUA..."
        />
      </div>
      <div
        class="file-upload-actions"
      >
        <button
          class="sds--btn sds--btn--primary  sds--btn--small  "
          type="button"
        >
          <span
            class="textual"
          >
            Change image
          </span>
        </button>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
